### ISSUE TITLE
Why use new_tensor() to make copy of student net's output in knowledgedistill?

### ISSUE THREAD

**Issue by nbadalls** [NervanaSystems/distiller#234]
Codes shows below
https://github.com/NervanaSystems/distiller/blob/8c5de42cf08e8d32b966a570c207039cb799a528/distiller/knowledge_distillation.py#L105-L125
  In line 123,  self.last_students_logits = out.new_tensor(out, requires_grad=True) use new_tensor() to create a copy of student network's output. new_tensor(x) is equival to x.clone().detch(), which means  to extract the output from the student graph, so that gradients could not directly work on parameters in student net. Why don't just use x.clone() instead?  
Looking forward your suggestion.

**Comment by guyjacob** [comment]
I don't recall a particular reason for implementing it that way, and it does look strange. Could definitely be a bug. I'll need to take a deeper look.

**Comment by nbadalls** [comment]
I see. My experiments also demonstrate that using clone() way gets better result compared to the new_tensor() way.
